llvm/lib/TargetParser/Triple.cpp

```diff
@@ -797,6 +798,16 @@ static Triple::SubArchType parseSubArch(StringRef SubArchName) {
   if (SubArchName == "arm64ec")
     return Triple::AArch64SubArch_arm64ec;
 
+  if (SubArchName.starts_with("spir")) {
+    StringRef SubArch(SubArchName);
+    if (SubArch.consume_front("spir64_") || SubArch.consume_front("spir_")) {
```

Comment:
we should consider if we also want to support `spirv64_gen`

@bader Opinion?

Reply:
This leads me to a separate question - given we should aways have an arch value when targeting GPU, can we get away with only using `spirv64` as the triple?  This would have the tools handle the proper target behavior when the arch value is encountered and not rely on the triple arch/subarch.  This transition from old to new model can also provide a clean break from using `spir64`.

Additional work would need to be done internally for existing usage of `spir64` and `spir64_gen` on build command lines and convert those accordingly.

Reply:
Yeah this is a good question, my understanding is AMD/NVIDIA don't support JIT at all, so they don't have to deal with AOT vs JIT.

I think this is a good topic for a discussion with a larger group, maybe the DPCPP technical forum?

Reply:
BTW what will the arch be for JIT?

Reply:
got it, then i definitely like the idea of triple being only `spirv64`, and the arch specifying the device/specific hw.

Reply:
> got it, then i definitely like the idea of triple being only `spirv64`, and the arch specifying the device/specific hw.

May be the 'arch' can be just empty for JIT. Why do we need 'generic'?

Thanks

Reply:
`generic` is the default value that is now filled for `arch=` for the packager.  This was introduced here: https://github.com/llvm/llvm-project/pull/126655